### ISSUE TITLE
Add lines of action rendering option

### ISSUE THREAD

**Issue by adamkewley** [ComputationalBiomechanicsLab/opensim-creator#562]
The muscle lines of action is used by some researchers to visualize muscles, but it requires a bit of fiddling around to get working. E.g.:

- https://twitter.com/HoonKimATC/status/1614616162909966336

(and, anecdotally, I have heard from other researchers that it's a nice visualization but requires a bit of work to get going)

The original plugin is hosted on SimTK:

- https://simtk.org/projects/force_direction/

And was developed by @modenaxe . I think it would be a nice visualization option to have in the next release of OpenSim Creator - provided it's easy enough to integrate, render, etc. in realtime.

The proposed integration point would be as a tickbox in this menu:

![Screenshot 2023-01-19 110033](https://user-images.githubusercontent.com/4730570/213418123-76412404-4d90-44e2-9c12-4fe6d2741ad7.png)

And, because it's a published visualization method, it should also have a tooltip that mentions the source paper, researcher, etc.

**Comment by adamkewley** [comment]
I seperately emailed @modenaxe about getting the source code. It might be that it's *somewhere*, but a basic search on github etc. didn't immediately yield anything.

If I can't get the source then I'll have a whack at reimplementing it from the source paper:

- https://pubmed.ncbi.nlm.nih.gov/23575923/

**Comment by adamkewley** [comment]
9db4fe3 contains a manually rewritten version of @modenaxe 's implementation. It's effectively the same algorithm but it skips a lot of the steps that the renderer doesn't need (e.g. flipping between coordinate spaces, writing to output files).

The effect looks pretty nice and it renders in realtime:

https://user-images.githubusercontent.com/4730570/213772318-e38bdbb0-b69e-449e-b41a-b0ecacfb20d7.mp4

But I need to check why the arrows sometimes dissapear on models containing wrapping surfaces. I suspect that there's an `if` check somewhere that's guarding the implementation from computing things.

**Comment by adamkewley** [comment]
This is now fully integrated.

It could do with some testing, refactors, etc. but that will be done while shipping 0.4.0, or by users who spot any oddities - it's not worth spending a large amount of time testing+streamlining this,  because it's an opt-in feature that's very nice to have but users can ignore it if they don't like it, or it's broken (hopefully, they'll report issues ;)).

**Comment by adamkewley** [comment]
(the latest commits are from `rebase`ing the commit onto `main` - to clean out the merge commits).

**Comment by adamkewley** [comment]
This is now merged into `main`, so it'll be used, tested, refactored, etc. as part of day-to-day 0.4.0 development.

Thanks again @modenaxe - I'll try to upstream the algorithms but I think it would be best to test this implementation in OSC for a while, get feedback on it (from a user PoV), etc. before pushing it up. 

One thing I noticed is that effective LoAs to "pop" when the muscle initially glides over a wrapping surface. Algorithmically, this makes sense, because as soon as the muscle glides over a wrapping surface new "virtual" (wrap) points are created and the algorithm will compute the "position" of the direction vector as being on the wrap point, rather than the anatomical point.

Because of that, the position "pops" onto the wrap surface. The direction vector moves smoothly though (good) - the pop might cause users to whine, though.
